packages/apps/docs/src/pages/api/blog.ts

```diff
@@ -0,0 +1,21 @@
+import { menuData } from '@/data/menu.mjs';
+import { getInitBlogPosts } from '@/hooks/useBlog/utils';
+import { IMenuData } from '@/types/Layout';
+import { NextApiRequest, NextApiResponse } from 'next';
+
+const search = async (
+  req: NextApiRequest,
+  res: NextApiResponse<IMenuData[] | IResponseError>,
+): Promise<void> => {
+  const { limit = 10, offset = 0 } = req.query as unknown as {
+    limit: number;
+    offset: number;
+  };
+
+  const data = getInitBlogPosts(menuData as IMenuData[], offset, limit);
+
+  res.status(200).json(data);
+  res.end();
```

Comment:
I'm pretty sure just `res.json()` inherently does `.status(200)` and also acts as `.end()` so those are both redundant.

Reply:
cool. lessons learned